### ISSUE TITLE
CA1861 does not work with C#12 array syntax

### ISSUE THREAD

**Issue by sungam3r** [dotnet/roslyn-analyzers#7090]
### Analyzer

**Diagnostic ID**: [CA1861](https://learn.microsoft.com/dotnet/fundamentals/code-analysis/quality-rules/ca1861): `Avoid constant arrays as arguments`

### Analyzer source

**SDK**: [Built-in CA analyzers in .NET 5 SDK or later](https://learn.microsoft.com/dotnet/fundamentals/productivity/code-analysis)

**Version**: [SDK 8.0.100](https://dotnet.microsoft.com/download/dotnet/8.0)

### Describe the bug

CA1861 does not work with C#12 array syntax.

### Steps To Reproduce

```xml
<Project Sdk="Microsoft.NET.Sdk">
  <PropertyGroup>
    <TargetFramework>net8.0</TargetFramework>
    <ImplicitUsings>enable</ImplicitUsings>
    <Nullable>enable</Nullable>
    <EnableNETAnalyzers>true</EnableNETAnalyzers>
    <TreatWarningsAsErrors>true</TreatWarningsAsErrors>
    <AnalysisMode>Recommended</AnalysisMode>
  </PropertyGroup>
</Project>
```

```cs
namespace ClassLibrary1
{
    public class Class1
    {
        public void Do()
        {
            DoSomething(new int[] { 1, 2 }); // CA1861
            DoSomething([1, 2]); // no CA1861
        }

        public static void DoSomething(int[] ints)
        {
        }
    }
}
```

### Expected behavior

CA1861 on both lines

### Actual behavior

CA1861 only for first line
![изображение](https://github.com/dotnet/roslyn-analyzers/assets/21261007/a61282c0-10c6-42df-b1b7-ba33d0ae0fb4)

**Comment by mavasani** [comment]
This would need us to add lightup operations for C# 12 collection literals

**Comment by manfred-brands** [comment]
> This would need us to add lightup operations for C# 12 collection literals

It is a pity that the CodeAnalysis is dependent on the SDK. We can't update references because it will conflict with the `Microsoft.CodeAnalysis` which is not shipped as a NuGet package but shipped as part of the SDK.
Unless we have separate versions for each SDK which is a nightmare.

Now every analyzer needs to add Lightup operations., this one, StyleCop does.
I have an open [issue](https://github.com/nunit/nunit.analyzers/issues/726) for NUnit.Analyzers for a similar issue.

If the dependency cannot be removed because the analyzer runs in the compiler address space , can the light up layer be a separate NuGet package so not every analyzer needs to make a new one?

Or instead of having one package, have one for each C# release.
Older SDKs won't load the new one and if an analyzer loads one the compiler doesn't know about it shouldn't conflict.

**Comment by manfred-brands** [comment]
> This would need us to add lightup operations for C# 12 collection literals

These seem to have been added 3 months ago for CA1870 (#7252)

**Comment by steveberdy** [comment]
That should be useful for the code fix as well. When I've used the code fix, it hasn't used the new syntax even with C# 12 enabled
